src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/HoverHandler.cs

```diff
@@ -0,0 +1,137 @@
+            LSPDocumentMappingProvider documentMappingProvider)
+        {
+            if (joinableTaskContext is null)
+            {
+                throw new ArgumentNullException(nameof(joinableTaskContext));
+            }
+
+            if (requestInvoker is null)
+            {
+                throw new ArgumentNullException(nameof(requestInvoker));
+            }
+
+            if (documentManager is null)
+            {
+                throw new ArgumentNullException(nameof(documentManager));
+            }
+
+            if (projectionProvider is null)
+            {
+                throw new ArgumentNullException(nameof(projectionProvider));
+            }
+
+            if (documentMappingProvider is null)
+            {
+                throw new ArgumentNullException(nameof(documentMappingProvider));
+            }
+
+            _joinableTaskFactory = joinableTaskContext.Factory;
+            _requestInvoker = requestInvoker;
+            _documentManager = documentManager;
+            _projectionProvider = projectionProvider;
+            _documentMappingProvider = documentMappingProvider;
+        }
+
+        public async Task<Hover> HandleRequestAsync(TextDocumentPositionParams request, ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
+        {
+            await _joinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            if (!_documentManager.TryGetDocument(request.TextDocument.Uri, out var documentSnapshot))
+            {
+                return null;
+            }
+
+            // Switch to a background thread.
+            await TaskScheduler.Default;
+
+            var projectionResult = await _projectionProvider.GetProjectionAsync(documentSnapshot, request.Position, cancellationToken).ConfigureAwait(false);
+            if (projectionResult == null)
+            {
+                return null;
+            }
+
+            var serverKind = projectionResult.LanguageKind == RazorLanguageKind.CSharp ? LanguageServerKind.CSharp : LanguageServerKind.Html;
+
+            if (serverKind != LanguageServerKind.CSharp)
+            {
+                // Hover is only supported in the C# LSP for now.
+                return null;
+            }
+
+            var textDocumentPositionParams = new TextDocumentPositionParams()
+            {
+                Position = projectionResult.Position,
+                TextDocument = new TextDocumentIdentifier()
+                {
+                    Uri = projectionResult.Uri
+                }
+            };
+
+            Hover result;
+            try
+            {
+                result = await _requestInvoker.RequestServerAsync<TextDocumentPositionParams, Hover>(
+                    Methods.TextDocumentHoverName,
+                    serverKind,
+                    textDocumentPositionParams,
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch
```

Comment:
Temporary blanket catch all to ensure silent failure (will be removed once Roslyn VS insertion is completed).

Reply:
File an issue to track removing this so we don't forget 😄

Reply:
https://github.com/dotnet/aspnetcore/issues/20962